Improve gRPC supported platforms doc (#21689)
dotnet/AspNetCore.Docs: aspnetcore/grpc/netstandard.md

.NET implementations that don't support HTTP/2, such as UWP, Xamarin and Unity, can use gRPC-Web as an alternative.

```csharp
var channel = GrpcChannel.ForAddress("https://localhost:5001", new GrpcChannelOptions
    {
        HttpHandler = new GrpcWebHandler(new HttpClientHandler())
    });

var client = new Greeter.GreeterClient(channel);
var response = await client.SayHelloAsync(new HelloRequest { Name = ".NET" });
```

For more information, see [Configure gRPC-Web with the .NET gRPC client](xref:grpc/browser#configure-grpc-web-with-the-net-grpc-client).

## .NET Framework

.NET Framework has limited support for gRPC over HTTP/2. To enable it, configure the channel to use <xref:System.Net.Http.WinHttpHandler>. This handler uses WinHTTP in Windows to send gRPC calls over HTTP/2.

There are requirements and restrictions to using `WinHttpHandler`:

* Ensure the operating system is Windows 10 Build 19622 or later.

> * Add a reference to the [System.Net.Http.WinHttpHandler](https://www.nuget.org/packages/System.Net.Http.WinHttpHandler/) package.

Reviewer: ```suggestion
* A reference to the [System.Net.Http.WinHttpHandler](https://www.nuget.org/packages/System.Net.Http.WinHttpHandler/) NuGet package.
```

* Only unary and server streaming gRPC calls are supported.
* Only gRPC calls over TLS are supported.

```csharp
var channel = GrpcChannel.ForAddress("https://localhost:5001", new GrpcChannelOptions
    {
        HttpHandler = new WinHttpHandler()
    });

var client = new Greeter.GreeterClient(channel);
var response = await client.SayHelloAsync(new HelloRequest { Name = ".NET" });
```

> [!NOTE]
> .NET Framework support is in its early stages and requires using pre-release software.
> * Windows 10 Build 19622 or later is available as a [Windows Insiders](https://insider.windows.com/) build.

> > * Required version of `System.Net.Http.WinHttpHandler` is not yet on NuGet.org. The latest pre-release version from the NuGet feed [here](https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json) should be used.

Reviewer: ```suggestion
> * The required version of `System.Net.Http.WinHttpHandler` is not currently available on NuGet.org. The latest pre-release version is available on [this NuGet feed](https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json) should be used.
```

## gRPC C# core-library

> An alternative option is to use [gRPC C# core-library](https://grpc.io/docs/languages/csharp/quickstart/) to make gRPC calls. It is a third party library that supports .NET Framework and Xamarin. gRPC C-core is not supported by Microsoft.

Reviewer: ```suggestion
An alternative option for .NET Framework is to use [gRPC C# core-library](https://grpc.io/docs/languages/csharp/quickstart/) to make gRPC calls. It's a third party library that supports .NET Framework and Xamarin. gRPC C-core is not supported by Microsoft.
```
